wagtail/search/tests/test_backends.py

```diff
@@ -428,6 +428,234 @@ def test_delete(self):
             "The Fellowship of the Ring"
         ])
 
+    #
+    # Basic query classes
+    #
+
+    def test_match_all(self):
+        results = self.backend.search(MATCH_ALL, models.Book.objects.all())
+        self.assertEqual(len(results), 13)
+
+    def test_term(self):
+        # Single word
+        results = self.backend.search(Term('Javascript'),
+                                      models.Book.objects.all())
+        self.assertSetEqual({r.title for r in results},
+                            {'JavaScript: The Definitive Guide',
+                             'JavaScript: The good parts'})
+
+        # Multiple word
+        results = self.backend.search(Term('Definitive Guide'),
```

Comment:
This shouldn't match anything as there "Definitive" and "Guide" are separate terms

Reply:
From a semantic point of view, I agree.
From a pragmatic point of view, I think it’s a bad idea to make it return nothing, for two reasons:
- This means we have to define a term separator, which is very far from being easy. This is usually up to the search engine to separate terms, and this is not a simple split by spaces. A few examples:
  - “rendez-vous” is a single term but “1980-2020” are two terms in most configuration
  - “can/must” are two terms but “github.com/wagtail/wagtail” is a single term
  - “Hello. Goodbye” are two terms but “127.0.0.1” is a single term
- To me, `Term` works a bit like a shell argument in `subprocess`. If you pass `Definitive Guide` as an argument to `subprocess`, it will not pass it as two separate argument, but as a single argument containing a space. Here, that’s exactly what we do. Additionally in the future, we may change that to a phrase query so it matches the order of words, like when searching `"Definitive Guide"` instead of `Definitive Guide` in Google.

Reply:
> To me, Term works a bit like a shell argument in subprocess. If you pass Definitive Guide as an argument to subprocess, it will not pass it as two separate argument, but as a single argument containing a space. Here, that’s exactly what we do. Additionally in the future, we may change that to a phrase query so it matches the order of words, like when searching "Definitive Guide" instead of Definitive Guide in Google.

That's exactly how I see it as well. But in the index, those two words would be split into separate terms, so the term query will not match either:

 - ``Definitive Guide``≠ ``Definitive``
 - ``Definitive Guide`` ≠ ``Guide``

When users are using this API, they should be aware about how the backend they are using is tokenising the text for their language. I don't think we should try to be clever here.

Reply:
Hmmm… Then maybe we’ll have a difference of behaviour between backends, here. In database backends, it doesn’t work this way. For example, in PostgreSQL if you do `to_tsquery('''Definitive Guide''')`, it returns the same query as `to_tsquery('Definitive & Guide')`.

Reply:
Isn't that the same as the ``PlainText`` query? Maybe we shouldn't support the term-level query types on database backends if they don't give that level of control.

Reply:
Yeah, and if we do that we should change as well `PlaintText`, as it splits terms in Python and provides an `operator` argument on which we have no control in `plainto_tsquery`. I made that splitting a bit quickly so it would easily work the same on all backends, but I’m not even sure that’s how to do it in Elasticsearch.

Reply:
For Elasticsearch, ``PlainText`` queries should map directly to a ``match`` query to avoid introducing any regressions (as that's what we do currently). ES also handles generating bigrams for CJK languages and could also have some user-specified behaviour defined in settings that we wouldn't want to break.

Reply:
It might be better long-term to support doing analysis on the Python side so all backends are analysed the same and it would be much easier to configure. But that's an issue for another day :)